producer/src/main/java/com/flipkart/varadhi/produce/services/ProducerService.java

```diff
@@ -236,19 +252,31 @@ private CompletableFuture<Result<Offset>> produceToStorageProducer(
         });
     }
 
-    private boolean applyOrgFilter() {
-        // TODO[IMP]: apply org filters
-        //        Project project = projectService.getCachedProject(projectName);
-        //        TODO[IMP]:avoid zk interactions for org and topic + filters
-        //        OrgFilters orgFilters = orgService.getAllFilters(project.getOrg());
-        //        VaradhiTopic topic = varadhiTopicService.get(buildTopicName(projectName, topicName));
-        //        String nfrStrategy = topic.getNfrFilterName();
-        //        Condition condition = (orgFilters != null && !orgFilters.getFilters().isEmpty()) ?
-        //                orgFilters.getFilters().get(nfrStrategy) :
-        //                null;
-        //        if (nfrStrategy != null && condition != null && condition.evaluate(message.getHeaders())) {
-        //            return true;
-        //        }
-        return false;
+    private boolean applyOrgFilter(String projectName, String topicName, Message message) {
+        Optional<Project> projectOptional = projectCache.get(projectName);
+        if (projectOptional.isEmpty()) {
```

Comment:
we can go with nested if ladder for this. 

if not project_optional.empty() {
    // remainder of the logic here
    if not projectOptional.empty() {
         // 
    }
}
return false;

Reply:
nested if is usually not recommended. flatten code is better to avoid nesting